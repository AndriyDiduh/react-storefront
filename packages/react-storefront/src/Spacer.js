/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component } from 'react'
import withStyles from '@material-ui/core/styles/withStyles'

/**
 * Renders a simple div with flex: 1 to be used as a spacer.  Since this is a
 * common case, the main purposed of this class is to minimize the amount of
 * css generated for an app.
 */
export const styles = theme => ({
  root: {
    flex: 1,
  },
})

@withStyles(styles, { name: 'RSFSpacer' })
export default class Spacer extends Component {
  render() {
    return <div className={this.props.classes.root} />
  }
}
