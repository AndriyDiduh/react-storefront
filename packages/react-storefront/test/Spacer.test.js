/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import Spacer from '../src/Spacer'

describe('Spacer', () => {
  it('renders', () => {
    const component = <Spacer />
    expect(mount(component)).toMatchSnapshot()
  })
})
