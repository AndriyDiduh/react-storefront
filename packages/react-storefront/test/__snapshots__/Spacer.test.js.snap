// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer renders 1`] = `
<WithStyles(Spacer)>
  <Spacer classes={{...}}>
    <div className="RSFSpacer-root-1" />
  </Spacer>
</WithStyles(Spacer)>
`;
